gcloud-java-compute/src/main/java/com/google/gcloud/compute/Snapshot.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine snapshot. Compute Engine allows you to take snapshots of your persistent
+ * disk and create new persistent disks from that snapshot. This can be useful for backing up data,
+ * recreating a persistent disk that might have been lost, or copying a persistent disk. Snapshots
+ * can be applied across persistent disk types. {@code Snapshot} adds a layer of service-related
+ * functionality over {@link SnapshotInfo}. Obejcts of this class are immutable, to get a
+ * {@code Snapshot} object with the most recent information use {@link #reload}.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/disks/persistent-disks#snapshots">Use
+ *     persistent disk snapshots</a>
+ */
+public class Snapshot extends SnapshotInfo {
+
+  private static final long serialVersionUID = -973924811396336695L;
+
+  private final ComputeOptions options;
+  private transient Compute compute;
+
+  /**
+   * A builder for {@code Snapshot} objects.
+   */
+  public static class Builder extends SnapshotInfo.Builder {
```

Comment:
In my understanding, snapshots (the server-side concept) are immutable (since there doesn't seem to be a "replace" or "update").  Is the primary purpose for having a Builder class/toBuilder() method so that users can take an existing snapshot, modify some fields, and then create a new snapshot?  There's probably value here (the user doesn't have to re-specify the sourceDisk/description, and it would be weird if calling toBuilder() on a `Snapshot` returned a `SnapshotInfo.Builder` rather than a `Snapshot.Builder`), but I just want to make sure I'm on the same page.